components/feature/downloads/src/main/java/mozilla/components/feature/downloads/SimpleDownloadDialogFragment.kt

```diff
@@ -0,0 +1,100 @@
+        val fileName = arguments?.get(KEY_FILE_NAME).toString()
+
+        alertDialogBuilder?.apply {
+            setupPositiveButtonListener()
+            return create()
+        }
+        return getDefaultDialog(fileName)
+    }
+
+    private fun getDefaultDialog(fileName: String): AlertDialog {
+        return AlertDialog.Builder(testingContext ?: requireContext())
+            .setTitle(dialogTitleText)
+            .setMessage(fileName)
+            .setNegativeButton(defaultNegativeButtonText) { _, _ ->
+                defaultNegativeButtonListener()
+            }
+            .setupPositiveButtonListener()
+            .create()
+    }
+
+    private fun AlertDialog.Builder.setupPositiveButtonListener(): AlertDialog.Builder {
+        this.apply {
+            setPositiveButton(positiveButtonText) { _, _ ->
+                positiveButtonListener()
+                onStartDownload()
+            }
+        }
+        return this
+    }
+
+    companion object {
+        /**
+         * A builder method for creating a [SimpleDownloadDialogFragment]
+         * @property dialogBuilder a way to provide your custom setup for the dialog and
+         * customize it to your wish. If [dialogBuilder] is not provide a default
+         * dialog will be created with negative and positive buttons.
+         * Be aware, you shouldn't provide a setPositiveButton, this behavior will be override.
+         * If you want have a listener for the positiveButton, you must use [positiveButtonListener].
+         * @property positiveButtonText the text that will be displayed on the positive button.
+         * @property positiveButtonListener this will be triggered when a positive button is clicked.
+         * @property defaultNegativeButtonText the text that will be displayed for the default negative button.
+         * If you provide a [dialogBuilder] this won't be applied.
+         * @property defaultNegativeButtonListener this will be triggered when the default negative
+         * If you provide a [dialogBuilder] this won't be called.
+         */
+        fun newInstance(
+            dialogBuilder: AlertDialog.Builder? = null,
+            @StringRes positiveButtonText: Int = mozac_feature_downloads_dialog_download,
+            positiveButtonListener: OnClickListener = {},
+            @StringRes defaultNegativeButtonText: Int = mozac_feature_downloads_dialog_cancel,
+            defaultNegativeButtonListener: OnClickListener = {}
+        ): SimpleDownloadDialogFragment {
+            val fragment = SimpleDownloadDialogFragment()
+            fragment.alertDialogBuilder = dialogBuilder
+            fragment.positiveButtonText = positiveButtonText
+            fragment.positiveButtonListener = positiveButtonListener
+            fragment.defaultNegativeButtonText = defaultNegativeButtonText
+            fragment.defaultNegativeButtonListener = defaultNegativeButtonListener
```

Comment:
If those are not in the arguments bundle .. doesn't that mean they will all get reset to the default when the fragment is re-created by Android?

Reply:
You are right, I'm putting them in the arguments bundle.